decorator.rb

```diff
@@ -0,0 +1,28 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
+end
+
+class Decorator < Nameable
+  def initialize(nameable)
+    @nameable = nameable
+    super()
```

Comment:
- [ ] **[OPTIONAL]** Adding super call here is not necessary since `Nameable` class does not have a constructor.